src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs

```diff
@@ -40,14 +208,18 @@ private sealed class CategoricalMetaData
             public bool[] IsCategoricalFeature;
         }
 
-        private protected readonly Options LightGbmTrainerOptions;
+        // Contains the passed in options when the API is called
+        private protected readonly TOptions LightGbmTrainerOptions;
+
+        // Contains the GBMOptions
```

Comment:
>GBMOptions [](start = 24, length = 10)

Ill move the argument below for this, I had rename to GbmOptions as it was conflicting with the Options class in the derived classes.